modules/repository-url/src/test/java/org/elasticsearch/repositories/url/RepositoryURLClientYamlTestSuiteIT.java

```diff
@@ -35,5 +57,59 @@ public RepositoryURLClientYamlTestSuiteIT(@Name("yaml") ClientYamlTestCandidate
     public static Iterable<Object[]> parameters() throws Exception {
         return ESClientYamlSuiteTestCase.createParameters();
     }
+
+    @Before
+    public void registerRepositories() throws IOException {
+        Response clusterSettingsResponse = client().performRequest("GET", "/_cluster/settings?include_defaults=true" +
+            "&filter_path=defaults.path.repo,defaults.repositories.url.allowed_urls");
+        Map<String, Object> clusterSettings = entityAsMap(clusterSettingsResponse);
+
+        @SuppressWarnings("unchecked")
+        List<String> pathRepo = (List<String>) XContentMapValues.extractValue("defaults.path.repo", clusterSettings);
+        assertThat(pathRepo, hasSize(1));
+
+        // Create a FS repository using the path.repo location
+        Response createFsRepositoryResponse = client().performRequest("PUT", "_snapshot/repository-fs", emptyMap(),
+            buildRepositorySettings(FsRepository.TYPE, Settings.builder().put("location", pathRepo.get(0)).build()));
+        assertThat(createFsRepositoryResponse.getStatusLine().getStatusCode(), equalTo(RestStatus.OK.getStatus()));
+
+        // Create a URL repository using the file://{path.repo} URL
+        Response createFileRepositoryResponse = client().performRequest("PUT", "_snapshot/repository-file", emptyMap(),
+            buildRepositorySettings(URLRepository.TYPE, Settings.builder().put("url", "file://" + pathRepo.get(0)).build()));
+        assertThat(createFileRepositoryResponse.getStatusLine().getStatusCode(), equalTo(RestStatus.OK.getStatus()));
+
+        // Create a URL repository using the http://{fixture} URL
+        @SuppressWarnings("unchecked")
+        List<String> allowedUrls = (List<String>) XContentMapValues.extractValue("defaults.repositories.url.allowed_urls", clusterSettings);
+        for (String allowedUrl : allowedUrls) {
+            try {
+                InetAddress inetAddress = InetAddress.getByName(new URL(allowedUrl).getHost());
+                if (inetAddress.isAnyLocalAddress() || inetAddress.isLoopbackAddress()) {
+                    Response createUrlRepositoryResponse = client().performRequest("PUT", "_snapshot/repository-url", emptyMap(),
+                        buildRepositorySettings(URLRepository.TYPE, Settings.builder().put("url", allowedUrl).build()));
+                    assertThat(createUrlRepositoryResponse.getStatusLine().getStatusCode(), equalTo(RestStatus.OK.getStatus()));
+                    break;
+                }
+            } catch (Exception e) {
+                logger.debug("Failed to resolve inet address for allowed URL [{}], skipping", allowedUrl);
```

Comment:
This seems like a pretty bad thing though. maybe we should throw it?

Reply:
This is required as the integration tests also declare `http://snapshot.test*` as an allowed url for some test, so I think we can keep it like this.